akka-projection-core/src/test/scala/akka/projection/internal/metrics/OffsetCommittedCounterMetricSpec.scala

```diff
@@ -0,0 +1,225 @@
+        }
+      }
+      "count offsets only once in case of failure" in {
+        val single = Handlers.singleWithErrors(1, 1, 3, 4, 5, 5, 5, 5, 6)
+        val tt = new TelemetryTester(
+          // using retryAndFail to try to get all message through
+          ExactlyOnce(recoveryStrategy = Some(HandlerRecoveryStrategy.retryAndFail(maxRetries, 30.millis))),
+          SingleHandlerStrategy(single))
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+        }
+      }
+    }
+    " in `exactly-once` with groupedHandler" must {
+      "count offsets" in {
+        val grouped = Handlers.grouped
+        val groupHandler = GroupedHandlerStrategy(grouped, afterEnvelopes = Some(2))
+        val tt =
+          new TelemetryTester(ExactlyOnce(), groupHandler)
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+          instruments.onOffsetStoredInvocations.get should be(3)
+        }
+      }
+      "count offsets only once in case of failure" in {
+        val groupedWithFailures = Handlers.groupedWithErrors(1, 2, 3)
+        val tt = new TelemetryTester(
+          ExactlyOnce(recoveryStrategy = Some(HandlerRecoveryStrategy.retryAndFail(maxRetries, 30.millis))),
+          GroupedHandlerStrategy(groupedWithFailures, afterEnvelopes = Some(2)))
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+        }
+      }
+    }
+
+  }
+
+}
+
+class OffsetCommittedCounterMetricAtMostOnceSpec extends InternalProjectionStateMetricsSpec {
+
+  val instruments = InMemInstruments
+
+  "A metric counting offsets committed" must {
+
+    // at-most-once
+    " in `at-most-once` with singleHandler" must {
+      "count offsets" in {
+        val tt =
+          new TelemetryTester(ExactlyOnce(), SingleHandlerStrategy(Handlers.single))
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+          instruments.onOffsetStoredInvocations.get should be(6)
+        }
+      }
+      "count offsets once in case of failure (skip)" in {
+        val single = Handlers.singleWithErrors(2, 3, 4)
+        val tt = new TelemetryTester(
+          AtMostOnce(recoveryStrategy = Some(HandlerRecoveryStrategy.skip)),
+          SingleHandlerStrategy(single))
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+        }
+      }
+      "count offsets once in case of failure (fail)" in {
+        val single = Handlers.singleWithErrors(4, 5, 6)
+        val tt = new TelemetryTester(
+          AtMostOnce(recoveryStrategy = Some(HandlerRecoveryStrategy.fail)),
+          SingleHandlerStrategy(single))
+
+        runInternal(tt.projectionState) {
+          instruments.offsetsSuccessfullyCommitted.get should be(6)
+        }
+      }
+    }
```

Comment:
I didn't do a 100% exhaustive test suite for all possible combinations. 
I added the `skip` and `fail` subcases here while tracking a bug on my implementation and ended up leaving both tests here to prevent regressions.